### PR TITLE
Updates and Modal UI for Website

### PR DESCRIPTION
**Pull request by Ememobong28**
**Description:**
This pull request adds user authentication and modal UI enhancements to the internship component of our application. With this update, users are now required to log in before accessing the internships section. If a user who is not logged in tries to access the internships page, a modal will appear prompting them to either log in or sign up. Additionally, there has been minor enhancement in the like feature.

**Milestones:**
1.) User authentication: This ensures that only authenticated users can access the internships page. 
2.) Modal UI: The addition of the modal enhances the user experience by providing a seamless login/signup prompt.
3.) Like Feature Enhancement: The minor tweaks in the like feature contribute to better user interactivity within the internships page. (This feature will be modified more in the future)

**Test Plan:**
1.) Navigate to the internships page without being logged in. Verify that the login/signup modal appears.
2.) Verify that the 'Close', 'Log in', and 'Sign up' buttons on the modal work as expected.
3.) After logging in, verify that you can access the internships page.
4.) Interact with the like button on the internship cards and ensure it behaves as expected.
5.) Lastly, run any existing unit tests and end-to-end tests to ensure the application works as expected with the new changes.

**Video Recordings**
[Screen Recording 2023-07-13 at 11.11.18 PM.mov.zip](https://github.com/Ememobong28/Capstone-Project-TechHive/files/12046824/Screen.Recording.2023-07-13.at.11.11.18.PM.mov.zip)

### CODE REVIEW

File: techhive-ui/src/components/Internships/Internships.css [hunk 1]
```diff
@@ -2,40 +2,42 @@
     padding: 20px;
     font-family: montserrat;
   }
-  
+
   .internship-cards {
     display: grid;
-    grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
-    gap: 20px;
+    grid-template-columns: 1fr; /* Change this to 1fr */
```
Comment (nicolec2626) [inline techhive-ui/src/components/Internships/Internships.css:8]: why do we need this comment?

File: techhive-ui/src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -20,15 +21,22 @@ function Navbar() {
...
-        ) : (
-          <li><Link to="/login">Login</Link></li>
-        )}
-        <li><Link to="/signup">SignUp</Link></li>
       </ul>
+      <div className="auth-container">
+        {user && <FaUserCircle className="profile-icon"/>}
+        <div className="auth-links">
+          {user ? (
+            <span onClick={handleLogout}>
```
Comment (nicolec2626) [inline techhive-ui/src/components/Navbar/Navbar.jsx:29]: where is handleLogout defined? I don't see it imported or defined in this file

**Review by nicolec2626 — COMMENTED** [review]
PR content looks much better with the detailed description + test plan!
 
the only nit I have is to make the PR title a little more specific: something like "Add user authentication and other improvements for internships page"

File: techhive-ui/src/components/Internships/Internships.css (hunk 1 above)
Comment (Ememobong28) [reply techhive-ui/src/components/Internships/Internships.css:8]: I was reminding myself to change it, I forgot to take it out

**Comment by Ememobong28** [comment]
> PR content looks much better with the detailed description + test plan!
> 
> the only nit I have is to make the PR title a little more specific: something like "Add user authentication and other improvements for internships page"

Okay noted, I will take note of that

File: techhive-ui/src/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (Ememobong28) [reply techhive-ui/src/components/Navbar/Navbar.jsx:29]: I defined it in the Logout.jsx file and also defined it in the navbar file. I don't know why you don't see it but it was there but when i modified i realized that I shouldn't have two logout so i took out handle logout from navbar and implemented it in Logout function and Imported it in navbar. It is in my updated code which I will be sending in.

**Pull request merged**